test/test_framework.py

```diff
@@ -328,26 +338,26 @@ def test_custom_event_data(self):
         framework = self.create_framework()
 
         class MyEvent(ops.EventBase):
-            def __init__(self, handle, n):
+            def __init__(self, handle: ops.Handle, n: int):
                 super().__init__(handle)
                 self.my_n = n
 
             def snapshot(self):
```

Comment:
Out of interest, why doesn't Pyright complain about this not having a return type signature?

Reply:
A lot of the type it can infer the return type. Possibly there's also some `__init__` special-casing, in the same way that `self` gets special-cased to not need `: typing.Self`?